src/components/List_Products.js

```diff
@@ -1,53 +1,80 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+import { Link, Outlet } from 'react-router-dom';
 import { onSnapshot } from 'firebase/firestore';
 
-import { productsCollection, saveItem } from '../lib/api';
+import { getListFromDB } from '../lib/api';
+import './ListProducts.css';
 
 function ListProducts() {
   const [items, setItems] = useState([]);
   const [itemName, setItemName] = useState('');
+  const list = useRef({});
```

Comment:
why are you guys using a `useRef` instead of a `useState`?

Reply:
The _list_ refers to the db document identified through the token. Since once the List component renders the first time **there's no way the list the user is seeing changes** (I mean, it can mutate -add, delete, update items- but the token and accordingly the list it identifies stay the same), and **because of this it won't trigger a re-render**, I decided not to use state. What do you think, would it be a better choice?

Reply:
Either should work, I just don't see it as something that should be more complex since it is only used for the name of the list. that was my doubt :)

Reply:
Oh I understand, I'll keep it in mind :)